src/Classes/Daemons/MyRadio_TrackAndTraceDaemon.php

```diff
@@ -0,0 +1,97 @@
+use MyRadio\MyRadioEmail;
+use MyRadio\ServiceAPI\MyRadio_List;
+
+class MyRadio_TrackAndTraceDaemon extends \MyRadio\MyRadio\MyRadio_Daemon
+{
+    public static function isEnabled()
+    {
+        return Config::$d_TrackAndTrace_enabled;
+    }
+
+    public static function run()
+    {
+        $weekkey = __CLASS__ . '_last_run_weekly';
+        if (self::getVal($weekkey) > time() - 604700) {
+            return;
+        }
+
+        self::generateTrackAndTraceReport();
+
+        // Done
+        self::setVal($weekkey, time());
+    }
+
+    private static function generateTrackAndTraceReport()
+    {
+        $table = "<table>";
+        $table .= "<tr><th>Type</th><th>Information</th><th>Location</th><th>Time</th></tr>";
+
+        $data = [];
+        $no_track = MyRadio_Timeslot::getLocationName(5); //WebStudio
+
+        foreach (MyRadio_Season::getAllSeasonsInLatestTerm() as $season) {
+            foreach ($season->getAllTimeslots() as $timeslot) {
+                if ($timeslot->getStartTime() < time()
+                    && $timeslot->getStartTime() > time() - 604800
+                ) {
+                    foreach ($timeslot->getSigninInfo() as $info) {
+                        if (isset($info["location"]) && $info["location"] != $no_track) {
+                            if (isset($info["user"])) {
+                                $eduroam = $info["user"]->getEduroam();
+                                $data[] = [
+                                    "type" => "URY Member",
+                                    "info" => $info["user"]->getName() . ($eduroam ? " ($eduroam)" : ""),
+                                    "location" => $info["location"],
+                                    "time" => CoreUtils::happyTime($info["time"]),
+                                    "unix" => $info["time"]
+                                ];
+                            } elseif ($info["guest_info"]) {
+                                $data[] = [
+                                    "type" => "Guest",
+                                    "info" => [
+                                        "display" => "html",
+                                        "html" => nl2br($info["guest_info"])
+                                    ],
+                                    "location" => $info["location"],
+                                    "time" => CoreUtils::happyTime($info["time"]),
+                                    "unix" => $info["time"]
+                                ];
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        usort($data, function ($a, $b) {
+            if ($a["unix"] == $b["unix"]) {
+                return 0;
+            }
+            return ($a["unix"] < $b["unix"]) ? -1 : 1;
+        });
+
+        foreach ($data as $row) {
+            $table .= "<tr><td>" . $row["type"]
+                . "</td><td>" . $row["info"]
```

Comment:
If it's a guest, `$row["info"]` will be a PHP array, I haven't tried it but I suspect it won't format nicely.

Reply:
the effort of dealing with that in a general case to the likelihood of needing to worry about it in an individual case isn't worth dealing with 😜

Reply:
Just tried it, and it doesn't even work:

```
php > echo "<td>" . ["html" => "testytest"] . "</td>";
PHP Notice:  Array to string conversion in php shell code on line 1
<td>Array</td>
```

Reply:
lol